### DISCUSSION TITLE
uninstall script not running on M1 migrated from Intel

### DISCUSSION THREAD

**Discussion by quickdry** [Homebrew/discussions#3437] [Getting started] [answered]
I migrated my old intel Mac onto an M1 (migration assistant, not timemachine). It brought over the intel homebrew installation and I'm trying to remove that and do a new fresh install.

The recommended: /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)"

Gives the error: 

/bin/bash: line 236: /opt/homebrew/bin/brew: No such file or directory
Failed to locate Homebrew!

which is what i'd expect as this has never had an M1 version of brew installed. I can uninstall all the packages and casks, but that still leaves brew. How do I remove that final piece of the puzzle properly so I can start fresh? (is ther a flag to force it to think it is only on an intel and not fail with the /opt/homebrew check?) The documentation I could find all seems to start from the POV of having no brew installation, or that the uninstall script just works.

**Comment by gromgit** [comment, marked as answer]
Try the following:
```
PATH=/usr/local/bin:$PATH arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)"
```
This "fakes out" the uninstaller in two ways:
1. Ensures that the Intel Homebrew path takes precedence, for when the uninstaller asks "which `brew` would normally be run?"
2. Ensures that the uninstaller is run in an Intel context

**Reply by quickdry** [reply]
worked like a charm, thank you

**Reply by zeerakminecraft** [reply]
/bin/bash: line 242: /usr/local/bin/brew: No such file or directory
Failed to locate Homebrew!

I am still getting this error. Kindly help

**Reply by SMillerDev** [reply]
What does `which brew` say?

**Reply by spyroseth** [reply]
I have the same error with @zeerakminecraft. My `which brew` outputs
```
which brew                                                                                                                                                                                                                            
/opt/homebrew/bin/brew
```

**Reply by SMillerDev** [reply]
In that case you don't have a rosetta brew install so there is nothing to uninstall.

**Reply by Tanish002** [reply]
@SMillerDev mine shows brew not found

**Reply by osalbahr** [reply]
@Tanish002 does https://github.com/orgs/Homebrew/discussions/775 help?

**Reply by Tanish002** [reply]
yes it worked thankyou so much

**Reply by jordanaf808** [reply]
I kept getting 
```
/bin/bash: line 242: /usr/local/bin/brew: No such file or directory
Failed to locate Homebrew!
```
After much research I was finally able to figure out that my Homebrew was _indeed_ not located there...
this is the line that worked for me
`/bin/bash uninstall.sh --path=/usr/local/Homebrew`
